pkg/dns/aws/dns.go

```diff
@@ -158,10 +158,10 @@ func NewProvider(config Config, operatorReleaseVersion string) (*Provider, error
 		// in the same region as the Route53 client to find the hosted zone
 		// of managed records.
 		tagConfig = tagConfig.WithRegion(endpoints.UsGovWest1RegionID)
-	case endpoints.UsIsoEast1RegionID:
-		// The resourcetagging API is not available in C2S
+	case endpoints.UsIsoEast1RegionID, endpoints.UsIsobEast1RegionID:
```

Comment:
I suggest that we take this opportunity to consider the AWS partition rather than the region. Otherwise, we are going to have to come back and make changes every time there are no regions added to the ISO or ISOB partitions.

Reply:
agreed. I can take a look how this is being managed in the installer and apply it here.

Reply:
Pushed new commit to switch from hard coded regions to partition based logic.